packages/synapse-sdk/src/telemetry/service.ts

```diff
@@ -17,13 +17,25 @@ import type { BrowserOptions, ErrorEvent, EventHint } from '@sentry/browser'
 import type { NodeOptions } from '@sentry/node'
 import type { FilecoinNetworkType } from '../types.ts'
 import { SDK_VERSION } from '../utils/sdk-version.ts'
-import { getSentry, isBrowser, type SentryBrowserType, type SentryType } from './utils.ts'
+import { getSentry, isBrowser, type SentryBrowserType, type SentryType, sanitizeUrlForSpan } from './utils.ts'
 
 type SentryInitOptions = BrowserOptions | NodeOptions
 type SentrySetTags = Parameters<SentryType['setTags']>[0]
 
 type SentryBeforeSendFunction = (event: ErrorEvent, hint: EventHint) => Promise<ErrorEvent | null>
 
+/**
+ * Extract the beforeSendSpan function type from BrowserOptions.
+ * This ensures we match Sentry's expected signature exactly.
+ */
+type SentryBeforeSendSpanFunction = NonNullable<BrowserOptions['beforeSendSpan']>
+
+/**
+ * Extract the span parameter type from beforeSendSpan.
+ * This is the JSON representation of a span that Sentry sends to the hook.
+ */
+type SpanJSON = Parameters<SentryBeforeSendSpanFunction>[0]
```

Comment:
I don't know if this is right or necessary.  I assume "SpanJSON" from https://docs.sentry.io/platforms/javascript/configuration/options/#beforeSendSpan is just a plain old JSON structure.

Reply:
This is actually invalid. Runtime values don't include `.tags` see this output:

```
modifiedSpan {
  data: {
    'sentry.source': 'custom',
    'sentry.sample_rate': 1,
    'sentry.op': 'payments.status',
    'sentry.origin': 'manual'
  },
  description: 'payments.status',
  op: 'payments.status',
  parent_span_id: undefined,
  span_id: 'c6099c1d32aefcc9',
  start_timestamp: 1762184458.675,
  status: 'ok',
  timestamp: 1762184461.3966377,
  trace_id: '462dd565c7b8e366776b2bed454f67eb',
  origin: 'manual',
  profile_id: undefined,
  exclusive_time: undefined,
  measurements: undefined,
  is_segment: true
}
```

Reply:
Per vebal, we are only adding an additional tag: url.sanitizedPath.

I'll resolve this.